package/InternalEAWGen.roc

```diff
@@ -0,0 +1,140 @@
+## The purpose of this file is to generate the InternalEAW.roc file.
+##
+## This file will read the test data from `data/EastAsianWidth-15.1.0.txt`
+## parse it and then generate function to test the East Asian Width property of a code point.
+app [main] {
+    pf: platform "https://github.com/roc-lang/basic-cli/releases/download/0.11.0/SY4WWMhWQ9NvQgvIthcv15AUeA7rAIJHAHgiaSHGhdY.tar.br",
+}
+
+import pf.File
+import pf.Task exposing [Task]
+import pf.Arg
+import "data/EastAsianWidth-15.1.0.txt" as file : Str
+import Helpers
+
+EawRange : (Str, Str, Str)
+
+main =
+    when Arg.list! |> List.get 1 is
+        Err _ -> Task.err (InvalidArguments "USAGE: roc run InternalEAWGen.roc -- path/to/package/")
+        Ok arg -> File.writeUtf8 "$(Helpers.removeTrailingSlash arg)/InternalEAW.roc" template
+
+template =
+    """
+    ## WARNING This file is automatically generated. Do not edit it manually. ##
+    module [eastAsianWidthProperty, eastAsianWidth]
+
+    eastAsianWidth : U32 -> U32
+    eastAsianWidth = \\codePoint ->
+        when eastAsianWidthProperty codePoint is
+            F | W | A -> 2
+            H | N | Na -> 1
+
+    eastAsianWidthProperty = \\cp -> $(testsStr)
+    """
+
+# The ranges are specified as its starting point and an optional inclusive end point
+# if it's length is greater than 1. The range is followed by the East_Asian_Width property
+# separated by a semicolon.
+# Examples:
+# 0000..001F     ; N    # The property may be followed by comments
+# 0020           ; Na
+parseLine : Str -> Result EawRange _
+parseLine = \line ->
+    listToParsingState = \rest -> { val: [], rest }
+
+    startHexBytes =
+        line
+        |> Str.toUtf8
+        |> listToParsingState
+        |> Helpers.takeHexBytes
+
+    endHexBytes =
+        when startHexBytes.rest is
+            ['.', '.', .. as rest] -> rest |> listToParsingState |> Helpers.takeHexBytes
+            _ -> startHexBytes # This range has length 1. The start will be repeated as the end
+
+    endHexBytes.rest
+    |> List.splitFirst ';'
+    |> Result.map \{ after } -> after
+    |> Result.try \commentdEAP -> List.splitFirst commentdEAP '#'
+    |> Result.map \{ before } -> before
+    |> Result.try Str.fromUtf8
+    |> Result.map Str.trim
+    |> Result.try \eawp ->
+        startHexBytes.val
+        |> Str.fromUtf8
+        |> Result.map \start -> (eawp, "0x$(start)")
+    |> Result.try \(eawp, start) ->
+        endHexBytes.val
+        |> Str.fromUtf8
+        |> Result.map \end -> (eawp, start, "0x$(end)")
+
+expect parseLine "0020           ; Na # Zs         SPACE" == Ok ("Na", "0x0020", "0x0020")
+expect parseLine "0025..0027     ; Na # Po     [3] PERCENT SIGN..APOSTROPHE" == Ok ("Na", "0x0025", "0x0027")
```

Comment:
Added a couple of unit tests, mostly to help check my understanding of this code.